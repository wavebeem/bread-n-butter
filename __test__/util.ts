import * as bnb from "../src/bread-n-butter";

export function snapTest<A>(parser: bnb.Parser<A>, input: string): void {
  expect(parser.parse(input)).toMatchSnapshot(JSON.stringify(input));
}
