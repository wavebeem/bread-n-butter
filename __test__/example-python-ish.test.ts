import Python from "../examples/python-ish";
import { snapTest } from "./util";

test("py complex", () => {
  snapTest(
    Python,
    `\
block:
  alpha
  bravo
  block:\r
          charlie
          delta\r
          echo
          block:
           foxtrot
  golf\
`
  );
});

test("py simple", () => {
  snapTest(
    Python,
    `\
block:
  alpha
  bravo
`
  );
});

test("py even simpler", () => {
  snapTest(Python, `alpha`);
});

test("py bad indent", () => {
  snapTest(
    Python,
    `\
block:
        alpha
        block:
    beta
`
  );
});
