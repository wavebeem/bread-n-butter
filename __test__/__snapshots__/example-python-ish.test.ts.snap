// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`py bad indent: "block:\\n    block:\\n        alpha\\n      beta\\n" 1`] = `
Object {
  "expected": Array [
    "6 spaces",
  ],
  "location": Object {
    "column": 7,
    "index": 38,
    "line": 4,
  },
  "type": "ParseFail",
}
`;

exports[`py complex: "block:\\n  alpha\\n  bravo\\n  block:\\r\\n          charlie\\n          delta\\r\\n          echo\\n          block:\\n           foxtrot\\n  golf" 1`] = `
Object {
  "type": "ParseOK",
  "value": Object {
    "statements": Array [
      Object {
        "type": "Ident",
        "value": "alpha",
      },
      Object {
        "type": "Ident",
        "value": "bravo",
      },
      Object {
        "statements": Array [
          Object {
            "type": "Ident",
            "value": "charlie",
          },
          Object {
            "type": "Ident",
            "value": "delta",
          },
          Object {
            "type": "Ident",
            "value": "echo",
          },
          Object {
            "statements": Array [
              Object {
                "type": "Ident",
                "value": "foxtrot",
              },
            ],
            "type": "Block",
          },
        ],
        "type": "Block",
      },
      Object {
        "type": "Ident",
        "value": "golf",
      },
    ],
    "type": "Block",
  },
}
`;

exports[`py even simpler: "alpha" 1`] = `
Object {
  "type": "ParseOK",
  "value": Object {
    "type": "Ident",
    "value": "alpha",
  },
}
`;

exports[`py simple: "block:\\n  alpha\\n  bravo\\n" 1`] = `
Object {
  "type": "ParseOK",
  "value": Object {
    "statements": Array [
      Object {
        "type": "Ident",
        "value": "alpha",
      },
      Object {
        "type": "Ident",
        "value": "bravo",
      },
    ],
    "type": "Block",
  },
}
`;
